lib/ZcapClient.js

```diff
@@ -0,0 +1,228 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+import {DEFAULT_HEADERS, httpClient} from '@digitalbazaar/http-client';
+import {SECURITY_CONTEXT_V2_URL} from 'jsonld-signatures';
+import {CapabilityDelegation} from 'ocapld';
+import {generateId} from 'bnid';
+import jsigs from 'jsonld-signatures';
+import {signCapabilityInvocation} from 'http-signature-zcap-invoke';
+const {Ed25519Signature2018} = jsigs.suites;
+
+export class ZcapClient {
+  /**
+   * An object that can make HTTPS requests such as node's https.Agent or ky.
+   *
+   * @typedef {object} httpsAgent
+   * @see https://nodejs.org/api/https.html#https_class_https_agent
+   *
+   * Creates a new ZcapClient instance that can be used to perform
+   * Authorization Capability (ZCAP) requests against HTTP URLs.
+   *
+   * @param {object} options - The options to use.
+   * @param {string} options.baseUrl - The base URL for the client to use
+   *   when building invocation request URLs.
+   * @param {object} [options.defaultHeaders] - The optional default HTTP
+   *   headers to include in every invocation request.
+   * @param {object} [options.agent] - An optional HttpsAgent to use to
```

Comment:
Is this what that typedef was for?

```suggestion
   * @param {HttpsAgent} [options.agent] - An optional HttpsAgent to use to
```